builtin/logical/ssh/path_config_ca.go

```diff
@@ -0,0 +1,63 @@
+package ssh
+
+import (
+	"fmt"
+	"github.com/hashicorp/vault/helper/errutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	"golang.org/x/crypto/ssh"
+)
+
+func pathConfigCA(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "config/ca",
+		Fields: map[string]*framework.FieldSchema{
+			"private_key": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: `Private half of the SSH key that will be used to sign certificates.`,
+			},
+			"public_key": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: `Public half of the SSH key that will be used to sign certificates.`,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.UpdateOperation: b.pathCAWrite,
+		},
+
+		HelpSynopsis: `Set the SSH private key used for signing certificates.`,
+		HelpDescription: `This sets the CA information used for certificates generated by this
+by this mount. The fields must be in the standard private and public SSH format.
+
+For security reasons, the private key cannot be retrieved later.`,
+	}
+}
+
+func (b *backend) pathCAWrite(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+
```

Comment:
We are already doing it for the private key. Its the public key that we could validate.